src/V3EmitCSyms.cpp

```diff
@@ -795,16 +792,27 @@ void EmitCSyms::emitSymImp() {
             }
             puts(protect("__Vscope_" + it->second.m_scopeName) + ".varInsert(__Vfinal,");
             putsQuoted(protect(it->second.m_varBasePretty));
-            puts(", &(");
+
+            std::string varName{""};
             if (modp->isTop()) {
-                puts(protectIf(scopep->nameDotless() + "p", scopep->protect()));
-                puts("->");
+                varName += (protectIf(scopep->nameDotless() + "p", scopep->protect()));
+                varName += ("->");
             } else {
-                puts(protectIf(scopep->nameDotless(), scopep->protect()));
-                puts(".");
+                varName += (protectIf(scopep->nameDotless(), scopep->protect()));
+                varName += (".");
             }
-            puts(varp->nameProtect());
-            puts("), ");
+            varName += (varp->nameProtect());
+
+            if (varp->isParam()) {
+                puts(", reinterpret_cast<void *>(");
```

Comment:
With this compilation passes, but reading a value will cause a core dump.

Reply:
I can help and look into it, the reinterpret cast and void* tick a couple of undesirable boxes ;)

Reply:
Yes, that would be great, it's been a while since I touched some proper c++.

Reply:
Verilator actually emits the parameter as an integer constant, that does not exist as variable. I would suggest to add another function in verilated.h, like constInsert, that would split it right away.